distributed/scheduler.py

```diff
@@ -3517,6 +3517,12 @@ class Scheduler(SchedulerState, ServerNode):
     default_port = 8786
     _instances: ClassVar[weakref.WeakSet[Scheduler]] = weakref.WeakSet()
 
+    worker_ttl: float | None
+    idle_since: float | None
+    idle_timeout: float | None
+    no_workers_since: float | None  # Note: not None iff there are pending tasks
+    no_workers_timeout: float | None
```

Comment:
In the spirit of #8190, is there a reason to make these public?

Reply:
As a rule of thumb it makes sense to me to keep the variables that hold a public setting as public too.
I don't have a strong opinion on the two `_since` variables or the two `check_` methods, so I'll change them to private (without deprecation, if you agree that there shouldn't be anybody hooking into them).

Reply:
I'll change the pre-existing attributes in a follow-up PR though.

Reply:
> As a rule of thumb it makes sense to me to keep the variables that hold a public setting as public too.

👍, makes sense to me.